### PR TITLE
Feat create accordian

### PR DESCRIPTION
**Pull request by QuinnrElder**
### Description

- First, was to recreate a Sparkbox exercise in React.
- Second, was to practice Sparkbox Git-Workflow best practices.
- Third, was to practice accessibility in a React setting.

### Spec
Designs: [Build Accessibly with Progressive Enhancement](https://sparkbox.com/foundry/toggle_aria-expanded_value_with_JavaScript_for_website_accessibility)

### Validation
* [x] Wave accessibility passes with no errors
* [x] This PR has code changes, and our linters still pass.
* [x] This PR has new SCSS functions, mixins, or variables, so [Sass Tests](https://seesparkbox.com/foundry/how_and_why_we_unit_test_our_sass) were added or updated, and they pass.

---

### CODE REVIEW

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
+$theme-primary: #7337ef;
+$theme-secondary: #444;
+
+dt {
+margin: .5rem;
+}
+
+.accordion {
+	&__button {
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:9]: Tabs should be set to use 2 spaces.

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		
+		&[aria-expanded='true'] {
+			background-color: $theme-secondary;
+		}
+		
+		&[aria-expanded='true']::after {
+			transform: rotate(315deg);
+		}
+	}
+}
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:50]: You'll want to update your editor's config to add an empty line at the bottom of each file.

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss [hunk 3]
```diff
@@ -0,0 +1,50 @@
+$theme-primary: #7337ef;
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:1]: I see these variables repeating in several partials which opens us up to things getting missed in the event that we want to update the primary color. We'll want to find a way to DRY up these variables and have them live in a single location. We experiment with importing the SCSS variables or using CSS custom properties.

Something else to consider is that we typically use [ITCSS](https://www.xfive.co/blog/itcss-scalable-maintainable-css-architecture/) to structure our styles, especially in non-react projects. All variables, SCSS or custom properties, live in a partial or partials that are imported at the top of the base.scss. This assumes that styles are not scoped to components like you have here. You can see a really good example of this in practice with documentation in the [sparkbox.com scss](https://github.com/sparkbox/seesparkbox.com/blob/main/src/scss/_base.scss)

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+$theme-secondary: #444;
+
+dt {
+margin: .5rem;
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:5]: You'll want to include a linter that will catch things such as the missing indent here. We have a custom [stylelint module](https://www.npmjs.com/package/@sparkbox/stylelint-config-sparkbox) that we use to get started and will [customize](https://github.com/sparkbox/seesparkbox.com/blob/main/.stylelintrc.js) if needed.

We also have one for [eslint](https://www.npmjs.com/package/@sparkbox/eslint-config-sparkbox).

File: sparkbox_best_practices/src/MockData/MockData.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
+export const PracticeDetails = {
+  accordianOne: {
+    accordianBarName: "What resources did you use to build this?",
+    linkInformation: [{
```
Comment (kaseybon) [inline sparkbox_best_practices/src/MockData/MockData.ts:4]: It's a little nit-picky but I would drop the curly brace to a new line. Same thing on lines 19, 56, and 71.

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss [hunk 6]
```diff
@@ -0,0 +1,50 @@
+$theme-primary: #7337ef;
+$theme-secondary: #444;
+
+dt {
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:4]: Avoid styling elements in your component styles. If we look at ITCSS all element styles should be styled only once if needed and they should scoped to the element section. We should make a class for this element using BEM or we can create a utility class system for spacing.

Comment (kaseybon) [reply sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:4]: You don't have an example of this in your code but you'll also want to [avoid nesting element selectors](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css) in classes.

🚫 `.accordion dt {…}`
✅ `.accordion__title  {…}`

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+margin: .5rem;
+}
+
+.accordion {
+	&__button {
+		background-color: $theme-primary;
+		color: #fff;
+		border-radius: 0.25rem;
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:12]: In most sites a log of these styles would be reused across other components and we'll want to assign them to variables. This includes the font family on line 19.

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import React, { Fragment, FunctionComponent } from 'react';
+import "./AccordianBar_.scss";
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.tsx:2]: You have a typo for accordion in a few spots.

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+			border-top: 0.25rem solid #fff;
+			transform: rotate(135deg);
+		}
+		
+		&:hover {
+			background-color: #444;
+			transform: scale(1.025);
+		}
+		
+		&[aria-expanded='true'] {
```
Comment (kaseybon) [inline sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:42]: This was a piece of feedback that was accidentally deleted but I wanted to make sure it was added back in.

I love this method of targeting the open/active state of an element. It helps to enforce accessibility by requiring the correct ARIA attributes are on the element.

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss (hunk 2 above)
Comment (QuinnrElder) [reply sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:50]: Fixed ✅

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss (hunk 1 above)
Comment (QuinnrElder) [reply sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:9]: Fixed ✅

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss (hunk 3 above)
Comment (QuinnrElder) [reply sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.scss:1]: Examples:
:root {
--background-primary: Green;
}

.accordion {
--background-primary: Blue;
}

File: sparkbox_best_practices/src/MockData/MockData.ts (hunk 5 above)
Comment (QuinnrElder) [reply sparkbox_best_practices/src/MockData/MockData.ts:4]: Fixed ✅

File: sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.tsx (hunk 8 above)
Comment (QuinnrElder) [reply sparkbox_best_practices/src/Components/AccordianBar_/AccordianBar_.tsx:2]: Fixed ✅

**Pull request merged**